servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/AbstractOpenTelemetryFilter.java

```diff
@@ -16,27 +16,49 @@
 
 package io.servicetalk.opentelemetry.http;
 
+import io.servicetalk.concurrent.PublisherSource;
+import io.servicetalk.concurrent.SingleSource;
+import io.servicetalk.concurrent.api.Publisher;
+import io.servicetalk.concurrent.api.Single;
+import io.servicetalk.concurrent.api.SourceAdapters;
 import io.servicetalk.http.api.HttpExecutionStrategies;
 import io.servicetalk.http.api.HttpExecutionStrategy;
 import io.servicetalk.http.api.HttpExecutionStrategyInfluencer;
+import io.servicetalk.http.api.StreamingHttpResponse;
 
-import io.opentelemetry.api.OpenTelemetry;
-import io.opentelemetry.api.trace.Tracer;
-import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
 
 abstract class AbstractOpenTelemetryFilter implements HttpExecutionStrategyInfluencer {
     static final OpenTelemetryOptions DEFAULT_OPTIONS = new OpenTelemetryOptions.Builder().build();
     static final String INSTRUMENTATION_SCOPE_NAME = "io.servicetalk";
-    final Tracer tracer;
-    final ContextPropagators propagators;
-
-    AbstractOpenTelemetryFilter(final OpenTelemetry openTelemetry) {
-        this.tracer = openTelemetry.getTracer(INSTRUMENTATION_SCOPE_NAME);
-        this.propagators = openTelemetry.getPropagators();
-    }
 
     @Override
     public final HttpExecutionStrategy requiredOffloads() {
         return HttpExecutionStrategies.offloadNone();
     }
+
+    static Single<StreamingHttpResponse> withContext(Single<StreamingHttpResponse> response, Context context) {
+        return new Single<StreamingHttpResponse>() {
+            @Override
+            protected void handleSubscribe(SingleSource.Subscriber<? super StreamingHttpResponse> subscriber) {
+                try (Scope ignored = context.makeCurrent()) {
+                    SourceAdapters.toSource(response.map(resp ->
```

Comment:
`response.map` operator here is unnecessary, you can directly call `response.transformMessageBody(...)`

Reply:
Maybe I'm misunderstanding your suggestion, but `response` is of type `Single<StreamingHttpResponse> response` so we can't operate on the Single directly.

Reply:
Oh right... Maybe name it `responseSingle` then?

Reply:
Good idea. 👍